docker/context/context.py

```diff
@@ -0,0 +1,208 @@
+import os
+import json
+from shutil import copyfile, rmtree
+from docker.tls import TLSConfig
+from docker.errors import ContextException
+from docker.context.config import get_meta_dir
+from docker.context.config import get_meta_file
+from docker.context.config import get_tls_dir
+from docker.context.config import get_context_host
+
+
+class Context:
+    """A context."""
+    def __init__(self, name, orchestrator="swarm", host=None, endpoints=None):
+        if not name:
+            raise Exception("Name not provided")
+        self.name = name
+        self.orchestrator = orchestrator
+        if not endpoints:
+            default_endpoint = "docker" if (
+                orchestrator == "swarm"
+                ) else orchestrator
+            self.endpoints = {
+                default_endpoint: {
+                    "Host": get_context_host(host),
+                    "SkipTLSVerify": False
+                }
+            }
+        else:
+            for k, v in endpoints.items():
+                ekeys = v.keys()
+                for param in ["Host", "SkipTLSVerify"]:
+                    if param not in ekeys:
+                        raise ContextException(
+                            "Missing parameter {} from endpoint {}".format(
+                                param, k))
+            self.endpoints = endpoints
+
+        self.tls_cfg = {}
+        self.meta_path = "IN MEMORY"
+        self.tls_path = "IN MEMORY"
+
+    def set_endpoint(
+            self, name="docker", host=None, tls_cfg=None,
+            skip_tls_verify=False, def_namespace=None):
+        self.endpoints[name] = {
+            "Host": get_context_host(host),
+            "SkipTLSVerify": skip_tls_verify
+        }
+        if def_namespace:
+            self.endpoints[name]["DefaultNamespace"] = def_namespace
+
+        if tls_cfg:
+            self.tls_cfg[name] = tls_cfg
+
+    def inspect(self):
+        return self.__call__()
+
+    @classmethod
+    def load_context(cls, name):
+        name, orchestrator, endpoints = Context.load_meta(name)
+        if name:
+            instance = cls(name, orchestrator, endpoints=endpoints)
+            instance.load_certs()
+            instance.meta_path = get_meta_dir(name)
+            return instance
+        return None
+
+    @classmethod
+    def load_meta(cls, name):
```

Comment:
This looks like an internal method. If yes, please prefix that with "_" to make that explicit